module/zfs/spa.c

```diff
@@ -3739,6 +3748,49 @@ spa_activity_check(spa_t *spa, uberblock_t *ub, nvlist_t *config)
 	return (error);
 }
 
+/*
+ * Called from zfs_ioc_clear for a pool that was suspended
+ * after failing mmp write checks.
+ */
+boolean_t
+spa_mmp_remote_host_activity(spa_t *spa)
+{
+	ASSERT(spa_multihost(spa) && spa_suspended(spa));
+
+	nvlist_t *best_label;
+	uberblock_t best_ub;
+	uberblock_t *spa_ub = &spa->spa_uberblock;
+	uint16_t spa_mmp_seq = MMP_SEQ_VALID(spa_ub) ? MMP_SEQ(spa_ub) : 0;
+
+	/*
+	 * Locate the best uberblock on disk
+	 */
+	vdev_uberblock_load(spa->spa_root_vdev, &best_ub, &best_label);
+	if (best_label)
+		nvlist_free(best_label);
+
+	/*
+	 * confirm that the best sequence is still from our host
+	 */
+	if (spa_mmp_seq != (MMP_SEQ_VALID(&best_ub) ? MMP_SEQ(&best_ub) : 0))
+		return (B_TRUE);
+
+	/*
+	 * confirm that the best hostid matches our host
+	 */
+	if (nvlist_exists(best_label, ZPOOL_CONFIG_HOSTID) &&
+	    spa_get_hostid(spa) !=
+	    fnvlist_lookup_uint64(best_label, ZPOOL_CONFIG_HOSTID)) {
+		return (B_TRUE);
+	}
```

Comment:
These checks should always be sufficient, but since we can't be too careful why don't we verify two more bits of state that MUST be true or we're somehow in some impossible state.

1. `MMP_VALID(best_ub)`, and
2. `MMP_FAIL_INT_VALID(best_ub) && MMP_FAIL_INT(best_ub) > 0`

both must be true, or the pool could never have been suspended on this node in the first place.  But there's no harm checking!

Reply:
good idea.  will add
```
+       if (!MMP_VALID(&best_ub) ||
+           !MMP_FAIL_INT_VALID(&best_ub) ||
+           MMP_FAIL_INT(&best_ub) == 0) {
+               return (B_TRUE);
+       }
```